src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
 export interface Figure {
-
+  shape: string,
+  color: string,
+  getArea: () => number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || c <= 0 || b <= 0) {
+      throw new Error('wrong side(s)');
+    }
+
+    const checkLongest = [this.a, this.b, this.c]
+      .sort((side1, side2) => side2 - side1);
+
+    if (checkLongest[0] >= checkLongest[1] + checkLongest[2]) {
+      throw new Error('unappropriated side');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const result = Math.sqrt(s * (s - this.a)
+      * (s - this.b) * (s - this.c));
 
+    return Number(result.toFixed(2));
+  }
 }
 
 export class Circle {
+  shape: string = 'circle';
 
-}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('wrong radius');
+    }
+  }
 
-export class Rectangle {
+  getArea(): number {
+    const result = Math.PI * (this.radius * this.radius);
 
+    return Math.round(result * 100) / 100;
```

Comment:
You can use Math.floor to fix bag whis wrong circle's square

Reply:
yeah, it worked, thanks